enterprise/cmd/repo-updater/internal/authz/perms_syncer_worker_test.go

```diff
@@ -50,21 +60,105 @@ func TestPermsSyncerWorker_Handle(t *testing.T) {
 			Priority:         database.LowPriorityPermissionSync,
 		})
 
-		wantMeta := &requestMeta{
-			ID:       4567,
-			Priority: priorityLow,
-			Type:     requestTypeRepo,
+		wantRequest := combinedRequest{
+			RepoID:  4567,
+			NoPerms: false,
 			Options: authz.FetchPermsOptions{
 				InvalidateCaches: false,
 			},
 		}
-
-		if diff := cmp.Diff(dummySyncer.request.requestMeta, wantMeta); diff != "" {
+		if diff := cmp.Diff(dummySyncer.request, wantRequest); diff != "" {
 			t.Fatalf("wrong sync request: %s", diff)
 		}
 	})
 }
 
+func TestPermsSyncerWorker(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	logger := logtest.Scoped(t)
+	db := database.NewDB(logger, dbtest.NewDB(logger, t))
+	ctx := context.Background()
+
+	// Creating users and repos.
+	userStore := db.Users()
+	user1, err := userStore.Create(ctx, database.NewUser{Username: "user1"})
+	require.NoError(t, err)
+	user2, err := userStore.Create(ctx, database.NewUser{Username: "user2"})
+	require.NoError(t, err)
+	repoStore := db.Repos()
+	err = repoStore.Create(ctx, &types.Repo{Name: "github.com/soucegraph/sourcegraph"}, &types.Repo{Name: "github.com/soucegraph/about"})
+	require.NoError(t, err)
+
+	// Creating a worker.
+	observationCtx := &observation.TestContext
+	dummySyncer := &dummySyncerWithErrors{
+		repoIDErrors: map[api.RepoID]struct{}{2: {}},
+		userIDErrors: map[int32]struct{}{2: {}},
+	}
+
+	workerStore := MakeStore(observationCtx, db.Handle())
+	worker := MakeTestWorker(ctx, observationCtx, workerStore, dummySyncer)
+	go func() { worker.Start() }()
+	t.Cleanup(func() { worker.Stop() })
```

Comment:
```suggestion
	go worker.Start()
	t.Cleanup(worker.Stop)
```

First-class functions, baby!

Reply:
lol :facepalm.jpg: